src/Illuminate/Foundation/Bootstrap/HandleExceptions.php

```diff
@@ -102,15 +102,15 @@ public function handleDeprecationError($message, $file, $line, $level = E_DEPREC
 
         $options = static::$app['config']->get('logging.deprecations') ?? [];
 
-        with($logger->channel('deprecations'), function ($log) use ($message, $file, $line, $level, $options) {
-            if ($options['trace'] ?? false) {
-                $log->warning((string) new ErrorException($message, 0, $level, $file, $line));
-            } else {
-                $log->warning(sprintf('%s in %s on line %s',
-                    $message, $file, $line
-                ));
-            }
-        });
+        $log = $logger->channel('deprecations');
+
+        if ($options['trace'] ?? false) {
+            $log->warning((string) new ErrorException($message, 0, $level, $file, $line));
+        } else {
+            $log->warning(sprintf('%s in %s on line %s',
+                $message, $file, $line
+            ));
+        }
```

Comment:
`Logger` uses the `Conditionable` trait, so we could do this:
```suggestion
        $logger->channel('deprecations')->when(
            $options['trace'] ?? false,
            fn ($log) => $log->warning((string) new ErrorException($message, 0, $level, $file, $line),
            fn ($log) => $log->warning(sprintf('%s in %s on line %s',
                $message, $file, $line
            ));
```